weatherv3/www/js/controllers.js

```diff
@@ -0,0 +1,116 @@
+      $scope.summary = current.currently.summary.toLowerCase();
+      $scope.tempDif = Math.abs(Math.round(current.currently.temperature) - Math.round(current.currently.apparentTemperature));
+      $scope.windSpeed = Math.round(current.currently.windSpeed);
+      var intensity = current.currently.precipIntensity;
+
+      $scope.precipIntensity = 0;
+      if (intensity == 0) {
+        $scope.precipIntensity = 0;
+      } else if (intensity < .002) {
+        $scope.precipIntensity = "very lightly";
+      } else if (intensity < .017) {
+        $scope.precipIntensity = "lightly";
+      } else if (intensity < .1) {
+        $scope.precipIntensity = "moderately hard";
+      } else {
+        $scope.precipIntensity = heavily;
+      }
+      $scope.precipProbability = current.currently.precipProbability;
+      $scope.precipType = current.currently.precipType;
+      $scope.nearestStormDistance = current.currently.nearestStormDistance;
+
+      console.log('tempDif: ', $scope.tempDif);
+      console.log('GOT CURRENT', current);
+      //debugger;
+    }, function(error) {
+      alert('Unable to get current conditions');
+      console.error(error);
+    });
+
+})
+
+.controller('weekCtrl', function($scope,$state,$filter,Weather,DataStore,WeekDays) {
+    //read default settings into scope
+    console.log('inside weekCtrl');
+    $scope.city  = DataStore.city;
+    var latitude  =  DataStore.latitude;
+    var longitude = DataStore.longitude;
+
+    //call getCurrentWeather method in factory ‘Weather’
+    Weather.getCurrentWeather(latitude,longitude).then(function(resp) {
+      $scope.current = resp.data;
+      console.log('GOT CURRENT', $scope.current);
+      theForecast();
+      //debugger;
+    }, function(error) {
+      alert('Unable to get current conditions');
+      console.error(error);
+    });
+
+    var weekDays = WeekDays;
+    var today = WeekDays.getToday();
+    //returns an array with info for the next 7 days
+    var theForecast = function() {
+      console.log('in theForecast function. value of current: ', $scope.current);
+      var d = new Date();
+      result = new Array(7);
+      var n = today;
+
+      $scope.barTypes = ['bar-calm', 'bar-balanced', 'bar-energized', 'bar-assertive', 'bar-royal', 'bar-dark','bar-positive', 'bar-calm', 'bar-balanced', 'bar-energized', 'bar-assertive', 'bar-royal', 'bar-dark','bar-positive'];
+      $scope.barFirst = n;
+      for (i = 0; i < 7; i++) {
+          var ctempMax = "High: " + Math.round($scope.current.daily.data[i].temperatureMax) + "°F, ";
+          var ctempMin = "Low: " + Math.round($scope.current.daily.data[i].temperatureMin) + "°F, ";
+          var cweatherSummary = $scope.current.daily.data[i].summary.toLowerCase();
+          
+          var cday = WeekDays[n] + ",";
+          var cdate = $filter('date')(d);
+          result[i] = {date: cdate, tempMax: ctempMax, tempMin: ctempMin, weatherSummary:cweatherSummary, day:cday};
+          
```

Comment:
Probably not good practice, but I needed new variable names so I put a c before each so they would stand out from the controller.